### PR TITLE
added sprite options to Keras Tensorboard Callback

### PR DESCRIPTION
**Pull request by beasteers**
I added `embeddings_image_path` and `embeddings_image_size` so that keras users can use the sprite feature with the embedding projector.

```python
class Tensorboard:
    def __init__(...
        embeddings_freq=0,
        embeddings_metadata=None,
        # added these
        embeddings_image_path=None,
        embeddings_image_size=None,
    )
```

Fixes issue #31050

### CODE REVIEW

**Comment by rchao** [comment]
@omalleyt12 adding you as you may have some context about TensorBoard callback, thanks

**Review by omalleyt12 — COMMENTED** [review]
Thanks for the PR! Could you add a unit test in tensorflow/python/keras/callbacks_test.py?

**Comment by beasteers** [comment]
>Could you add a unit test
Sure thing. It won't be in the next couple days though. 

First though, I think there needs to be a bit of discussion about how the embedding projector is expected to be used with the Tensorboard callback.

Judging by the changes to the TB callback in v2, it seems like it's only intended to work with word embeddings and support for image embeddings was pretty much dropped (image embeddings aren't generally used via `Embedding` layers, and `embedding_layer_names`/`embedding_data` were removed as parameters. Now it only gathers Embedding layers)

Here's an issue someone opened about the topic which describes the problem: https://github.com/tensorflow/tensorflow/issues/33230

**Comment by omalleyt12** [comment]
> Sure thing. It won't be in the next couple days though.

That works, thanks! Even just a simple test to show that the code with the new options sets runs and that the file it writes to is not empty would be fine (especially if you could attach an image to this PR showing how the resulting embeddings look on TensorBoard for a sample model)

> it seems like it's only intended to work with word embeddings and support for image embeddings was pretty much dropped

This wasn't intended, as long as the TensorBoard projector can support image embeddings we should support them in the TensorBoard callback. Our test coverage was likely too poor to catch this regression.

Does this PR make it possible to visualize image embeddings?

**Comment by beasteers** [comment]
>Does this PR make it possible to visualize image embeddings?

Not entirely. It supports image embeddings that make use of the Embedding layer, (but I'm not sure if that's even possible to do.) Making these changes in `v1` would though, but I'm not sure if we're supposed to make changes there.

Once we add `embedding_layer_names` (and `embedding_data`) back in we'll be able to visualize the output of a dense layer which is usually where image/non-discrete embeddings come from.

**Review by omalleyt12 — APPROVED** [review]
Thanks for the PR!

**Comment by gbaned** [comment]
@beasteers Can you please check build failures? Thanks!

**Comment by gbaned** [comment]
@beasteers Could you please address Ubuntu Sanity errors? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 43 days that this pull-request has stalled. Please create a new pull-request with the requested changes.
